library/alloc/src/sync.rs

```diff
@@ -1056,7 +1056,7 @@ impl<T: ?Sized> Arc<T> {
     // Non-inlined part of `drop`.
     #[inline(never)]
     unsafe fn drop_slow(&mut self) {
-        // Destroy the data at this time, even though we may not free the box
+        // Destroy the data at this time, even though we might not free the box
         // allocation itself (there may still be weak pointers lying around).
```

Comment:
i think the "may not" in line 1059 is a "must not"
(also change the "may" in line 1060 to "might"?)